### ISSUE TITLE
Model training stops all of a sudden

### ISSUE THREAD

**Issue by Ca-ressemble-a-du-fake** [DigitalPhonetics/IMS-Toucan#55]
Hi, 

Thank you very much for this amazing project! I run 2 trainings in parallel on an RTX 3090 (on a separate headless machine) because the memory footprint of a single training is low (around 10 GB) so there is room for 2 models!

I use `screen` to make it run even when I disconnect. But from time to time when I reload the screen sessions, one of the trainings is stopped in the middle of an epoch computation, and it shows `process interrupted` without error. This can happen some minutes after I launch the training, or some hours, after 12k steps or after 22k steps (spectrogram loss can be 0.18 or 0.31, not `NaN`). 

How can I find the cause of this interruption ? 

THanks in advance for your help

**Comment by Flux9665** [comment]
It might be because of torch/cuda trying to allocate ressources on the same GPU to different processes. Does the same also happen when you just train a single model on the GPU, even though there's a bit of wasted space? (you can of couse always just increase the batchsize to fill the space)

I use nohup to detach my processes and I never saw this issue, so it might also be linked to using screen.

**Comment by Ca-ressemble-a-du-fake** [comment]
Thank you @Flux9665. I have already been using screen to train 2 (CoquiTTS) models simultaneously for weeks without problem so screen should be OK. When one of the training stops, the other one keeps training, so I guess the problem would not appear on a single model training.

By the way I could not find a log file, is there such a file somewhere, or is there a way to make the training more verbose ? 

Regarding batchsize I have already increased it from the default 12 to 32, should I keep increasing it or on the contrary decrease it (I saw a conversation with [Michal Conrad](https://github.com/DigitalPhonetics/IMS-Toucan/issues/19) but there is no conclusion :smile:) ?

**Comment by Flux9665** [comment]
You can use the wandb.ai tracking service to track your experiments, you just need to be logged in and then add --wandb to the arguments of the call. All logs are written to std:out, I usually redirect std:out into a file when I call the training using nohup.

For FastSpeech the batchsize is not that important. For autoregressive models, you usually need a large batchsize, but non-autoregressive models are pretty robust to that. It's just another hyperparameter to try out empirically :)

**Comment by Ca-ressemble-a-du-fake** [comment]
That's very interesting thank you very much ! I was not aware of this insensitivity of Fastspeech2 model  to batch size.

The problem is that nothing happens on std:out when it stops except "processus interrompu" (in French). If there was an Out Of Memory error it would show that right (like Cuda error)? I will try to redirect std:out to a file maybe I am missing something ... 

By the way, today I watched the output of `nvidia-smi` and saw that one of the model I was  simultaneously training was now eating nearly 22 GB of VRAM (it was trained alone this time).  So an unnoticed Out Of Memory error must have been triggered the last time I tried to train 2 models concurrently. I'll try again to be sure with wandb enabled as you are suggesting.

In coqui there is an option to first train against the longest samples (so that you see shortly after starting the training if it fails. If I want to do that, would it work to sort the samples by descending order in `path_to_transcript_dicts`, or the preprocessing that comes afterwards will break that order  ? 

Thanks again for your prompt help

**Comment by Flux9665** [comment]
You can redirect std:out and std:err both into the same file to get the error messages.

Sorting by length and then adapting the batchsize to the length of items in the batch was something I wanted to do for a long time. The problem is that it's important to shuffle the data, which you can then only do in buckets. But this would require a custom dataloader and so far I'm just using the builtin one from torch. So all of the datapoints are shuffled, you cannot influence the order.
